### ISSUE TITLE
May I ask why the network loss nan

### ISSUE THREAD

**Issue by kweisamx** [twtygqyy/pytorch-SRDenseNet#2]
As I write, when we use very deep network, the output often be very big, that make loss nan,
why your network won't be nan?

Thanks.

**Comment by twtygqyy** [comment]
@kweisamx lot of network architecture tricks such as PReLu, Residual Block, and gradient clippings could help to avoid gradient vanishing.

**Comment by kweisamx** [comment]
Thank you for response, @twtygqyy ,
I know the gradient vanishing , But I want to know why it won't gradient explosion?
Do you use Batch Normalization?, I don't see something like BH in code

**Comment by twtygqyy** [comment]
Hi @kweisamx, please check out https://zh.coursera.org/learn/deep-neural-network/lecture/C9iQO/vanishing-exploding-gradients for theoretical details. In very deep network architectures, it is more likely to have gradient vanishing issue due to dead ReLU. 
Regarding Batch Normalization, BN would reduce the contrast of the image which is good for image classification while not good for super resolution. You can take a look at Instance Normalization which could be a better way, as has been pointed out in DeblurGAN (https://arxiv.org/pdf/1711.07064.pdf)

**Comment by kweisamx** [comment]
Hi @twtygqyy , Thanks for your answer.
Close
